networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/MultiaddrUtil.java

```diff
@@ -45,10 +48,33 @@ public static Multiaddr fromInetSocketAddress(
     return addPeerId(fromInetSocketAddress(address), nodeId);
   }
 
+  public static InetSocketAddress getResolvedInetSocketAddress(NetworkConfig config)
+      throws UnknownHostException {
+    final InetSocketAddress advertisedAddress =
+        new InetSocketAddress(config.getAdvertisedIp(), config.getAdvertisedPort());
+    final InetSocketAddress resolvedAddress;
+    if (advertisedAddress.getAddress().isAnyLocalAddress()) {
+      resolvedAddress =
+          new InetSocketAddress(InetAddress.getLocalHost(), advertisedAddress.getPort());
+    } else {
+      resolvedAddress = advertisedAddress;
+    }
+    return resolvedAddress;
+  }
+
+  public static Multiaddr fromInetSocketAddress(
+      final InetSocketAddress address, final byte[] nodeId) {
```

Comment:
We should avoid using `byte[]` for nodeId - it makes it very unclear what type of node ID it actually is (discv5 and libp2p have very different node IDs).  `NodeId` is an interface for that purpose but at the moment I have no idea what the spec actually wants for the discovery node ID.